### PR TITLE
Convert some more core classes to kotlin

### PR DESCRIPTION
**Pull request by keyboardsurfer**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing

### CODE REVIEW

File: core/src/main/java/io/plaidapp/core/data/api/DeEnvelopingConverter.kt [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    }
+                }
+                return@use null
+            }
+        }
+    }
+
+    private fun getPayloadName(annotations: Array<Annotation>?): String? {
+        if (annotations == null) return null
+        for (annotation in annotations) {
```
Comment (nickbutcher) [inline core/src/main/java/io/plaidapp/core/data/api/DeEnvelopingConverter.kt]: Use `find` here?

File: core/src/main/java/io/plaidapp/core/data/api/DeEnvelopingConverter.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import retrofit2.Converter
+import retrofit2.Retrofit
+import java.lang.reflect.Type
+import javax.inject.Inject
+
+/**
+ * A [retrofit2.Converter.Factory] which removes unwanted wrapping envelopes from API responses.
+ */
+class DeEnvelopingConverter @Inject
+constructor(internal val gson: Gson) : Converter.Factory() {
```
Comment (emartynov) [inline core/src/main/java/io/plaidapp/core/data/api/DeEnvelopingConverter.kt]: Is it recommended code style to move `constructor` to a new line?

Comment (keyboardsurfer) [reply core/src/main/java/io/plaidapp/core/data/api/DeEnvelopingConverter.kt]: It's not recommended but the autoformatter that picked up the annotation. Changed as it's all less than 100 chars.

File: core/src/main/java/io/plaidapp/core/data/pocket/PocketUtils.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    private const val PACKAGE = "com.ideashower.readitlater.pro"
+    private const val MIME_TYPE = "text/plain"
+    private const val EXTRA_SOURCE_PACKAGE = "source"
+    private const val EXTRA_TWEET_STATUS_ID = "tweetStatusId"
+
+    @JvmOverloads
+    fun addToPocket(
+        context: Context,
+        url: String?,
```
Comment (nickbutcher) [inline core/src/main/java/io/plaidapp/core/data/pocket/PocketUtils.kt]: Make this non-nullable?

Comment (keyboardsurfer) [reply core/src/main/java/io/plaidapp/core/data/pocket/PocketUtils.kt]: Not until PlaidItem is immutable.
Filed #795

**Pull request merged**